akka-stream/src/main/scala/akka/stream/scaladsl/Tcp.scala

```diff
@@ -328,26 +376,109 @@ final class Tcp(system: ExtendedActorSystem) extends akka.actor.Extension {
   }
 
   /**
-   * INTERNAL API
+   * Creates a [[Tcp.ServerBinding]] instance which represents a prospective TCP server binding on the given `endpoint`
+   * where all incoming and outgoing bytes are passed through TLS.
+   *
+   * You specify a factory to create an SSLEngine that must already be configured for
+   * client mode and with all the parameters for the first session.
+   *
+   * @see [[Tcp.bind]]
    */
-  @InternalApi private[akka] def bindTlsWithSSLEngine(
+  def bindWithTls(
+      interface: String,
+      port: Int,
+      createSSLEngine: () => SSLEngine): Source[IncomingConnection, Future[ServerBinding]] =
+    bindWithTls(
+      interface,
+      port,
+      createSSLEngine,
+      backlog = defaultBacklog,
+      options = Nil,
+      idleTimeout = Duration.Inf,
+      verifySession = _ => Success(()),
+      closing = IgnoreComplete)
+
+  /**
+   * Creates a [[Tcp.ServerBinding]] instance which represents a prospective TCP server binding on the given `endpoint`
+   * where all incoming and outgoing bytes are passed through TLS.
+   *
+   * You specify a factory to create an SSLEngine that must already be configured for
+   * client mode and with all the parameters for the first session.
+   *
+   * @see [[Tcp.bind]]
+   */
+  def bindWithTls(
       interface: String,
       port: Int,
       createSSLEngine: () => SSLEngine,
-      backlog: Int = 100,
-      @silent // Traversable deprecated in 2.13
-      options: immutable.Traversable[SocketOption] = Nil,
-      idleTimeout: Duration = Duration.Inf,
+      backlog: Int,
+      options: immutable.Seq[SocketOption],
+      idleTimeout: Duration,
       verifySession: SSLSession => Try[Unit],
-      closing: TLSClosing = IgnoreComplete): Source[IncomingConnection, Future[ServerBinding]] = {
+      closing: TLSClosing): Source[IncomingConnection, Future[ServerBinding]] = {
```

Comment:
I choose to not use any default param values in the new methods to make it easier to add overloads in the future, if needed